### PR TITLE
Welcome feature

### PR DESCRIPTION
**Pull request by norowachi**
ig i'm a github idiot

### CODE REVIEW

File: src/database/functions/GuildSettingsFunctions.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ function GuildScheme(guildID: string) {
             welcomeChannelMessage: "Welcome {member} to {server}!",
             dmMessage: "Welcome {member} to {server}!",
             welcomeRoles: [],
-            CustomWelcomeBackground: ""
```
Comment (july12123) [inline src/database/functions/GuildSettingsFunctions.ts]: if you change it here you also need to change it in schemas GuildSettingsSchema.ts and interfaces global types

File: src/database/functions/GuildSettingsFunctions.ts [hunk 2]
```diff
@@ -23,18 +23,19 @@ function GuildScheme(guildID: string) {
...
+            urlWhiteList: [],
         },
         welcomeSettings: {
             enable: false,
             welcomeDM: false,
             welcomeChannel: false,
             welcomeChannelID: "",
-            channelMessage: "Welcome {member} to {server}!",
-            dmMessage:"Welcome {member} to {server}!",
+            welcomeChannelMessage: "Welcome {member} to {server}!",
```
Comment (july12123) [inline src/database/functions/GuildSettingsFunctions.ts]: again you can't just change it in one place you need to change it in like 2 other files

File: src/database/functions/GuildSettingsFunctions.ts [hunk 3]
```diff
@@ -63,90 +64,165 @@ export async function removeGuildSettings(guild_id: string) {
...
+    if (data.welcomeDM == undefined) {
+        data.welcomeDM = Settings.welcomeDM;
+    }
+    if (data.welcomeChannel == undefined) {
+        data.welcomeChannel = Settings.welcomeChannel;
+    }
+    if (data.welcomeChannelID == undefined) {
+        data.welcomeChannelID = Settings.welcomeChannelID;
+    }
+    if (data.channelMessage == undefined) {
```
Comment (july12123) [inline src/database/functions/GuildSettingsFunctions.ts:96]:
```suggestion
    if (data.welcomeChannelMessage == undefined) {
```

File: src/database/functions/GuildSettingsFunctions.ts [hunk 4]
```diff
@@ -63,90 +64,165 @@ export async function removeGuildSettings(guild_id: string) {
...
+        data.channelMessage = Settings.channelMessage;
```
Comment (july12123) [inline src/database/functions/GuildSettingsFunctions.ts:97]:
```suggestion
        data.welcomeChannelMessage = Settings.welcomeChannelMessage;
```

File: src/database/functions/GuildSettingsFunctions.ts [hunk 5]
```diff
@@ -63,90 +64,165 @@ export async function removeGuildSettings(guild_id: string) {
...
-        if(data.welcomeRoles == undefined){data.welcomeRoles = Settings.welcomeRoles}
-
-        await guildSettings.updateOne({guild_id:guildID},{welcomeSettings:data})
+    guildID: string,
+    data: {
+        enable?: boolean;
+        welcomeDM?: boolean;
+        welcomeChannel?: boolean;
+        welcomeChannelID?: string;
+        channelMessage?: string;
```
Comment (july12123) [inline src/database/functions/GuildSettingsFunctions.ts:73]:
```suggestion
        welcomeChannelMessage?: string;
```

File: src/database/functions/GuildSettingsFunctions.ts (hunk 3 above)
Comment (rohank05) [reply src/database/functions/GuildSettingsFunctions.ts:96]: If(!data.something)

File: src/database/functions/GuildSettingsFunctions.ts (hunk 2 above)
Comment (norowachi) [reply src/database/functions/GuildSettingsFunctions.ts]: ay that, i forgot to change it to channelMessage

File: src/database/functions/GuildSettingsFunctions.ts (hunk 3 above)
Comment (july12123) [reply src/database/functions/GuildSettingsFunctions.ts:96]: no some items are bools and it will mess with it (experience first hand)

**Pull request merged**